ui-sdk-components-sample/UISDK components sample/UIKit Karhoo Component/ViewControllerPresenter.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  ViewControllerPresenter.swift
+//  UIKit Karhoo Component
+//
+//  Copyright © 2020 Karhoo. All rights reserved.
+//
+
+import Foundation
+import KarhooUISDK
+import KarhooSDK
+import SwiftSpinner
+
+protocol SampleViewControllerPresenter {
+    func didLoad(view: SampleView)
+    func didSelect(quote: Quote)
+}
+
+final class ViewControllerPresenter: SampleViewControllerPresenter {
+
+    var view: SampleView? = nil
+    private var bookingStatus = KarhooBookingStatus.shared
+
+    func didLoad(view: SampleView) {
+        self.view = view
+        bookingStatus.add(observer: self)
+        authenticateAndStartDemo()
+    }
+
+    /* ensure authentication before showing UI */
+    private func authenticateAndStartDemo() {
+        if Karhoo.configuration.authenticationMethod().guestSettings != nil {
+            view?.setUpView()
+        } else {
+            let userService = Karhoo.getUserService()
+
+            userService.login(userLogin: Keys.userLogin).execute(callback: { [weak self] result in
+                                                    switch result {
+                                                    case .success(_ ):
+                                                        self?.view?.setUpView()
+                                                    case .failure(let error):
+                                                        print("error! \(String(describing: error ?? nil))")
+                                                    }
+                                                   })
+        }
+    }
+
+    // QuoteList component output
+    func didSelect(quote: Quote) {
+        let bookingRequestScreen = KarhooUI()
+            .screens()
+            .bookingRequest()
+            .buildBookingRequestScreen(quote: quote,
+                                       bookingDetails: bookingStatus.getBookingDetails()!,
+                                       callback: { [weak self] result in
+                                        self?.handleBookingResult(result)
+                                       })
+        view?.presentView(viewController: bookingRequestScreen)
+    }
+
+    // BookingRequestScreen output
```

Comment:
Would be good to use the component instead

Reply:
There is no component for that, its a screen in iOS as it has a faded background. Components and screens are interchangeable depending on requirements :)